mdtraj/geometry/rdf.py

```diff
@@ -67,7 +70,10 @@ def compute_rdf(traj, pairs=None, r_range=None, bin_width=0.005,
         r_range = np.array([0.0, 1.0])
     r_range = ensure_type(r_range, dtype=np.float64, ndim=1, name='r_range',
                           shape=(2,), warn_on_cast=False)
-    bins = np.arange(r_range[0], r_range[1], bin_width)
+    if n_bins is not None:
```

Comment:
I wouldn't do an `assert`.  Maybe something like:

```Python
if n_bins is not None:
    n_bins = int(n_bins)
    if n_bins <= 0:
        raise ValueError('n_bins must be a positive integer')
```

`assert`s get removed when you compile Python code with `-OO` (docstrings are removed when compiled with `-O`).  So you should only ever use `assert` if, in the absence of any internal bug, program behavior will never change if that line is removed (because you never know when it will behind the scenes).  That's still Pythonic, I think.

Again, apologies if I took what you said too literally and I was rehashing what you already knew.